config/widget-antivirus/antivirus_status.widget.php

```diff
@@ -123,8 +104,16 @@ function dwg_av_statistic() {
 			<td class="vncellt">HTTP Scanner</td>
 			<td class="listr" width="75%">
 			<?php
-				// HAVP version; note - obviously broken now
-				echo exec("pkg_info | grep \"[h]avp\"");
+				// HAVP version
+				$pfs_version = substr(trim(file_get_contents("/etc/version")), 0, 3);
+				if ($pfs_version == "2.1") {
+					echo exec("pkg_info | grep \"[h]avp\"");
+				} elseif ($pfs_version == "2.2") {
+					// Show package version at least, no good quick way to get the PBI version
+					echo "pkg v{$config['installedpackages']['package'][get_pkg_id("havp")]['version']}";
+				} else {
+					echo exec("/usr/sbin/pkg info havp | /usr/bin/head -n 1");
```

Comment:
You can use pkg_exec() here to run 'pkg query "%v" havp' that will return only version